### ISSUE TITLE
RestAssured.requestSpecification not working with cookies

### ISSUE THREAD

**Issue by gaaroth** [rest-assured/rest-assured#680]
As discussed in this topic
https://groups.google.com/forum/#!topic/rest-assured/lczHl16q7K8

Trying to set a custom default requestSpecification in the RestAssured class doesn't carry over to next calls. Ex:
```
Cookies cookies = RestAssured.given().auth().basic("user", "password").get("/login").detailedCookies();
RequestSpecification myRequestSpecification = new RequestSpecBuilder().addCookies(cookies).build();
RestAssured.requestSpecification = myRequestSpecification ;
//any call after this doesn't use cookies
```
Calling .given().spec(myRequestSpecification) works of course but apparently the same .spec() called inside .given() (which makes a merge in the groovy code) doesn't seems to set everything properly.

Workaround that seems to make it work is:
```
RestAssured.requestSpecification = new RequestSpecBuilder().build();
RestAssured.given();  
Cookies cookies = RestAssured.given().auth().basic("user", "pass").get("/login").detailedCookies();
RestAssured.requestSpecification = new RequestSpecBuilder().addCookies(cookies).build();
//All calls after this use cookies
```
So basically creating an empty specification, calling .given() and then replacing with the real specification makes it work.
I'm available for any further question.
Regards!

**Comment by johanhaleby** [comment]
Hmm I actually can't reproduce it. I have the following test:

```java
RestAssured.requestSpecification = new RequestSpecBuilder().addCookie("my-cookie", "1234").build();

try {
    when().
            post("/reflect").
    then().
            log().ifValidationFails().
            cookie("my-cookie", "1234");
} finally {
    RestAssured.reset();
}
```

`reflect` only returns the cookies that were sent in the request in the response as well.

**Comment by johanhaleby** [comment]
Closing due to inactivity
